### PR TITLE
sql,streamingccl: introduce ALTER TENANT $1 COMPLETE REPLICATION

### PR DESCRIPTION
**Pull request by stevendanna**
This introduces

   ALTER TENANT <tenant name> COMPLETE REPLICATION TO LATEST
   ALTER TENANT <tenant name> COMPLETE REPLICATION TO SYSTEM TIME <time>

Which sets the cutover timestamp on the replication job. We do not yet add the ability to wait for the replication to complete.

The crdb_internal function has been left in place for now, but we may consider removing it in a follow-up PR.

Informs: #91265

Release note: None

### CODE REVIEW

**Comment by stevendanna** [comment]
I think the test failure so far has to do with a general problem in our tests that we can resolve now that we have the StartTime recorded.

Namely, in a lot of places we do:

1. get time
2. wait for high watermark to be past that time
3. cutover

But, the time chosen in 1 isn't necessarily a safe cutover time. It may be before the start time.

**Review by lidorcarmel — APPROVED** [review]
looks great, just a few nits about the errors.. but we can fix that later.

File: pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go [hunk 1]
```diff
@@ -369,6 +369,8 @@ func TestTenantStreamingSuccessfulIngestion(t *testing.T) {
 	require.Equal(t, "running the SQL flow for the stream ingestion job",
 		runningStatus(t, c.destSysSQL, ingestionJobID))
 
+	srcTime := c.srcCluster.Server(0).Clock().Now()
+	c.waitUntilHighWatermark(srcTime, jobspb.JobID(ingestionJobID))
```
[the remark this answers is not in the document]
Comment (stevendanna) [reply pkg/ccl/streamingccl/streamingest/stream_replication_e2e_test.go]: This was to make sure that we didn't attempt to set a cutover time before a protected timestamp had been recorded. But it was incomplete. I've replaced it with a function that waits for the frontier to reach the start time.

**Review by lidorcarmel — APPROVED** [review]
thanks, lgtm.

**Pull request merged**